anvil/src/window_map.rs

```diff
@@ -297,6 +300,29 @@ impl WindowMap {
         None
     }
 
+    pub fn bring_surface_to_top(&mut self, surface: &WlSurface) {
+        let found = self.windows.iter().enumerate().find(|(_, w)| {
+            if let Some(s) = w.toplevel.get_surface() {
+                s.as_ref().equals(surface.as_ref())
+            } else {
+                false
+            }
+        });
+
+        if let Some((i, _)) = found {
+            let winner = self.windows.remove(i);
+
+            // Take activation away from all the windows
+            for window in self.windows.iter() {
+                window.toplevel.set_activated(false);
+            }
+
+            // Give activation to our winner
+            winner.toplevel.set_activated(true);
+            self.windows.insert(0, winner);
+        }
+    }
```

Comment:
I'm not thrilled by the code duplication with `get_surface_and_bring_to_top`, but I'm not quite sure how to avoid it.

Reply:
I moved the "move the nth window to top" part to a separate method, that helps a little